### ISSUE TITLE
Language Server 'Folder' Scope settings not being considered.

### ISSUE THREAD

**Issue by NikolasKomonen** [Microsoft/vscode#69543]
Initially started discussion here: https://github.com/angelozerr/lsp4xml/pull/293#issuecomment-461104212

**The following is what I think I understand so far**

In a VSCode workspace there are 3 scopes for settings (User, Workspace, Folder).

A language server's package.json configuration.properties setting can have a scope of (application, window, and resource) with resource considering all 3 setting scopes with priority:  
`  Folder > Workspace > User ` ('>' is greater than in this case).

So for example a Folder scoped setting would override existing Workspace and User settings.

---

The issue I have is that it seems like Folder scoped settings are not being considered at all.

You can test this with any language server. But here are steps for reproducing on HTML:

1) Open an HTML file in a VSCode **Workspace**.
2) Go to Preferences and set `"html.trace.server"` to `verbose`
3) Also in Preferences for each setting scope (User, Workspace, Folder)
  set a different setting for [`"html.format.WrapAttributes"`](https://github.com/Microsoft/vscode/blob/master/extensions/html-language-features/package.json#L115)
4) View tab -> Output -> HTML Language Server from dropdown menu
5) ctrl+f and search for "wrapAttributes" where you should see the Workspace scoped preference being sent to the server, and not the Folder one.

**Comment by sandy081** [comment]
Assigning to @octref for investigating whether the configuration is being read properly in html extension

**Comment by aeschli** [comment]
I just tested that folder specific settings work (I tried with `"html.format.wrapLineLength"`).

How it works is that the language server *asks* for the configuration of a document whenever needed. (request: `workspace/configuration`). That way only the client needs to know about the various scopes. 
What you see being sent from the client to the server in `workspace/didChangeConfiguration` is no longer used, only as fallback in case the client doesn't support the `workspace/configuration` request.
